pkg/infrastructure/azure/network.go

```diff
@@ -0,0 +1,468 @@
+	inboundNatRulesClient := in.networkClientFactory.NewInboundNatRulesClient()
+	inboundNatRulesPoller, err := inboundNatRulesClient.BeginDelete(ctx,
+		in.resourceGroupName,
+		in.loadBalancerName,
+		in.inboundNatRuleName,
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to delete inbound nat rule: %w", err)
+	}
+	_, err = inboundNatRulesPoller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete inbound nat rule: %w", err)
+	}
+	return nil
+}
+
+func associateInboundNatRuleToInterface(ctx context.Context, in *inboundNatRuleInput) (*armnetwork.Interface, error) {
+	interfacesClient := in.networkClientFactory.NewInterfacesClient()
+	interfaceResp, err := interfacesClient.Get(ctx,
+		in.resourceGroupName,
+		in.bootstrapNicName,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get interface: %w", err)
+	}
+	bootstrapInterface := interfaceResp.Interface
+
+	inboundNatRulesClient := in.networkClientFactory.NewInboundNatRulesClient()
+	inboundNatRulesResp, err := inboundNatRulesClient.Get(ctx,
+		in.resourceGroupName,
+		in.loadBalancerName,
+		in.inboundNatRuleName,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get inbound nat rule: %w", err)
+	}
+	inboundNatRule := inboundNatRulesResp.InboundNatRule
+
+	for i, ipConfig := range bootstrapInterface.Properties.IPConfigurations {
+		ipConfig.Properties.LoadBalancerInboundNatRules = append(ipConfig.Properties.LoadBalancerInboundNatRules,
+			&inboundNatRule,
+		)
+		bootstrapInterface.Properties.IPConfigurations[i] = ipConfig
+	}
+
+	interfacesPollerResp, err := interfacesClient.BeginCreateOrUpdate(ctx,
+		in.resourceGroupName,
+		in.bootstrapNicName,
+		bootstrapInterface,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add inbound nat rule to interface: %w", err)
+	}
+
+	interfacesResp, err := interfacesPollerResp.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add inbound nat rule to interface: %w", err)
+	}
+	return &interfacesResp.Interface, nil
+}
+
+// GetCloudConfiguration gets a cloud configuration from the cloud name and endpoint.
+func GetCloudConfiguration(cloudName aztypes.CloudEnvironment, armEndpoint string) (*cloud.Configuration, error) {
+	cloudEnv, err := azureenv.EnvironmentFromURL(armEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var cloudConfig cloud.Configuration
+	switch cloudName {
+	case aztypes.StackCloud:
+		cloudConfig = cloud.Configuration{
+			ActiveDirectoryAuthorityHost: cloudEnv.ActiveDirectoryEndpoint,
+			Services: map[cloud.ServiceName]cloud.ServiceConfiguration{
+				cloud.ResourceManager: {
+					Audience: cloudEnv.TokenAudience,
+					Endpoint: cloudEnv.ResourceManagerEndpoint,
+				},
+			},
+		}
+	case aztypes.USGovernmentCloud:
+		cloudConfig = cloud.AzureGovernment
+	case aztypes.ChinaCloud:
+		cloudConfig = cloud.AzureChina
+	// XXX: What do we do here?
+	// case aztypes.GermanCloud:
```

Comment:
I don't think we need to worry about this as the German cloud was shutdown: https://github.com/Azure/azure-sdk-for-go/issues/20293#issuecomment-1446485961, https://www.azurespeed.com/Information/AzureSovereignClouds

Reply:
Fixed.